external_parser/event_processors/typed_events.h

```diff
@@ -39,11 +60,17 @@ template <> struct event_processor<v2::MultiSlotEvent> {
             evt.context()->size()};
   }
 
-  static joined_event::joined_event fill_in_joined_event(
-      const v2::MultiSlotEvent &evt, const v2::Metadata &metadata,
-      const TimePoint &enqueued_time_utc, std::string &&line_vec) {
+  static joined_event::joined_event
+  fill_in_joined_event(dsjson_metrics *metrics, const v2::MultiSlotEvent &evt,
+                       const v2::Metadata &metadata,
+                       const TimePoint &enqueued_time_utc,
+                       std::string &&line_vec) {
     auto ccb_data = VW::make_unique<joined_event::ccb_joined_event>();
 
+    if (metrics) {
+      calculate_metrics(metrics, metadata, enqueued_time_utc);
```

Comment:
no need for incrmenting zeroactions here?

Reply:
probably is but I want to tackle ccb in a separate PR